storybookjs/storybook
# Type safe select

Issue:

This solves the case in the `select`-knob when you pass in an array or an object where the values are not strings but the prop the knob is used for expects another type (e.g. a number), see for example issue #4106 

## What I did
The passed in options of the `select`-knob are stored in the local state, so they are not converted to strings when setting them as a value of the `option`-element. On a selection, the original value will be received from the state.

## How to test

Is this testable with Jest or Chromatic screenshots?
Does this need a new example in the kitchen sink apps? yes - see added story for testing.
Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
